This is a known issue upstream, with no fix:
http://comments.gmane.org/gmane.comp.video.ffmpeg.trac/3347

We don't need this code in debug so I'll just disable it for now.

diff -ruN libav-0.8.5/libavcodec/x86/cabac.h libav-0.8.5.new/libavcodec/x86/cabac.h
--- libav-0.8.5/libavcodec/x86/cabac.h	2013-01-12 09:00:42.000000000 -0800
+++ libav-0.8.5.new/libavcodec/x86/cabac.h	2013-02-19 09:14:31.694758334 -0800
@@ -100,6 +100,7 @@
 }
 #endif /* HAVE_7REGS && !defined(BROKEN_RELOCATIONS) */
 
+#if 0
 #define get_cabac_bypass_sign get_cabac_bypass_sign_x86
 static av_always_inline int get_cabac_bypass_sign_x86(CABACContext *c, int val)
 {
@@ -134,5 +135,6 @@
     );
     return val;
 }
+#endif // 0
 
 #endif /* AVCODEC_X86_CABAC_H */
diff -ruN libav-0.8.5/libpostproc/postprocess_template.c libav-0.8.5.new/libpostproc/postprocess_template.c
--- libav-0.8.5/libpostproc/postprocess_template.c	2013-01-12 09:00:42.000000000 -0800
+++ libav-0.8.5.new/libpostproc/postprocess_template.c	2013-02-19 09:15:50.598761406 -0800
@@ -1043,7 +1043,7 @@
 #if !HAVE_ALTIVEC
 static inline void RENAME(dering)(uint8_t src[], int stride, PPContext *c)
 {
-#if HAVE_MMX2 || HAVE_AMD3DNOW
+#if 0 //HAVE_MMX2 || HAVE_AMD3DNOW
     __asm__ volatile(
         "pxor %%mm6, %%mm6                      \n\t"
         "pcmpeqb %%mm7, %%mm7                   \n\t"
